### PR TITLE
cr/cs annotation interceptor for OkHttp

### PR DESCRIPTION
**Pull request by tburch**
This is the first part of #174; I've created an Interceptor for adding cr/cs annotations to OkHttp requests.

### CODE REVIEW

File: brave-okhttp/src/main/java/com/github/kristofa/brave/okhttp/OkHttpRequest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public String getHttpMethod() {
+        return getRequest().method();
+    }
+
+    @Override
+    public URI getUri() {
+        return getRequest().url().uri();
```
Comment (JakeWharton) [inline brave-okhttp/src/main/java/com/github/kristofa/brave/okhttp/OkHttpRequest.java]: It's wasteful to be calling `getRequest()` in these methods. Consider taking in the original `Request` and storing it in a field alongside the mutable `newBuilder()`

File: brave-okhttp/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Example of configuring the interceptor with OkHttp:
+
+```
+import com.github.kristofa.brave.okhttp.BraveOkHttpRequestResponseInterceptor;
+
+...
+
+OkHttpClient client = new OkHttpClient.Builder()
+    .addInterceptor(new BraveOkHttpRequestResponseInterceptor())
```
Comment (JakeWharton) [inline brave-okhttp/README.md]: This isn't exactly accurate as it takes some arguments. Maybe put something like `..` between the parens to keep them omitted.

File: brave-okhttp/src/main/java/com/github/kristofa/brave/okhttp/OkHttpRequest.java (hunk 1 above)
Comment (tburch) [inline brave-okhttp/src/main/java/com/github/kristofa/brave/okhttp/OkHttpRequest.java]: Thanks. I originally had it that way and then was trying to make it a little simpler. I'll go back to that.

**Comment by tburch** [comment]
@JakeWharton I made those two changes.

File: brave-okhttp/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# brave-okhttp #
+
+Contributed by [Tristan Burch](https://github.com/tburch).
```
Comment (adriancole) [inline brave-okhttp/README.md]: Please remove. while certainly appreciate the help, we dont typically do this (even if some crept in)

File: brave-okhttp/pom.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>brave</artifactId>
+    <groupId>com.github.kristofa</groupId>
+    <version>3.7.1-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>brave-okhttp</artifactId>
+  <name>brave-okhttp-interceptor</name>
```
Comment (adriancole) [inline brave-okhttp/pom.xml]: @kristofa interceptor or interceptors? Personally i like just plain brave-okhttp

File: brave-okhttp/pom.xml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.github.kristofa</groupId>
+      <artifactId>brave-http</artifactId>
+      <version>3.7.1-SNAPSHOT</version>
+    </dependency>
+    <dependency>
+      <groupId>com.squareup.okhttp</groupId>
+      <artifactId>mockwebserver</artifactId>
+      <version>2.7.5</version>
```
Comment (adriancole) [inline brave-okhttp/pom.xml]: Surprised theres no variable for this. Maybe okhttp.version?

File: brave-okhttp/src/test/java/com/github/kristofa/brave/okhttp/BraveOkHttpRequestResponseInterceptorTest.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+/**
```
Comment (adriancole) [inline brave-okhttp/src/test/java/com/github/kristofa/brave/okhttp/BraveOkHttpRequestResponseInterceptorTest.java]: Nuke ide auto stuff

File: brave-okhttp/pom.xml (hunk 5 above)
Comment (JakeWharton) [inline brave-okhttp/pom.xml]: Need to switch groupId to `com.squareup.okhttp3` in order to use `okhttp.version`.

File: brave-okhttp/README.md (hunk 3 above)
Comment (tburch) [inline brave-okhttp/README.md]: No Problem. Just copied the readme from the http client module and changed things.

File: brave-okhttp/pom.xml (hunk 4 above)
Comment (tburch) [inline brave-okhttp/pom.xml]: I like brave-okhttp as well. There will be another Interceptor coming...

File: brave-okhttp/src/test/java/com/github/kristofa/brave/okhttp/BraveOkHttpRequestResponseInterceptorTest.java (hunk 6 above)
Comment (tburch) [inline brave-okhttp/src/test/java/com/github/kristofa/brave/okhttp/BraveOkHttpRequestResponseInterceptorTest.java]: Will do.

File: pom.xml [hunk 7]
```diff
@@ -45,6 +45,7 @@
     <maven-source-plugin.version>2.4</maven-source-plugin.version>
     <maven-javadoc-plugin.version>2.10.3</maven-javadoc-plugin.version>
     <maven-gpg-plugin.version>1.6</maven-gpg-plugin.version>
+    <okhttp.version>3.2.0</okhttp.version>
```
Comment (kristofa) [inline pom.xml]: I would be in favor of keeping okhttp version and dependency in brave-okhttp/pom.xml. wdyt?

Comment (adriancole) [inline pom.xml]: good point

Comment (tburch) [inline pom.xml]: Made that change.

**Comment by tburch** [comment]
It seems like the tests in `brave-spancollector-scribe` will randomly pass/fail... I don't think this is related to the changes I've made.

**Comment by kristofa** [comment]
@tburch Agree. Looks like the `brave-spancollector-scribe` tests are flaky.

**Comment by adriancole** [comment]
kicked the build. will merge on green unless someone says nooooooooooooo

**Comment by kristofa** [comment]
👍

**Comment by tburch** [comment]
Thanks! 👍

**Comment by adriancole** [comment]
thanks for getting this started!

**Pull request merged**